packages/smooth_app/lib/helpers/picture_capture_helper.dart

```diff
@@ -1,39 +1,105 @@
+      // if task is greate than 6 , that means it has been executed 7 times
+      if (counter > 6) {
+        // returns true to let platform know that the task is completed
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
+      }
+      final List<Duration> duration = <Duration>[
+        const Duration(seconds: 30),
+        const Duration(minutes: 1),
+        const Duration(minutes: 30),
+        const Duration(hours: 1),
+        const Duration(hours: 6),
+        const Duration(days: 1),
+      ];
+      bool shouldRetry = false;
+      try {
+        final SendImage image = SendImage(
+          lang: ProductQuery.getLanguage(),
+          barcode: inputData['barcode'].toString(),
+          imageField:
+              ImageFieldExtension.getType(inputData['imageField'].toString()),
+          imageUri: Uri.parse(inputData['imageUri'].toString()),
+        );
+        final Status result = await OpenFoodAPIClient.addProductImage(
+          ProductQuery.getUser(),
+          image,
+        );
+        shouldRetry = result.error != null || result.status != 'status ok';
+      } catch (e) {
+        shouldRetry = true;
+      }
+      if (shouldRetry) {
+        inputData['counter'] = counter + 1;
+        await Workmanager().initialize(callbackDispatcher);
+        await Workmanager().registerOneOffTask(
+          task,
+          'ImageUploadWorker',
+          constraints: Constraints(
+            networkType: NetworkType.connected,
+          ),
+          inputData: inputData,
+          initialDelay: duration[counter],
+        );
+        return Future<bool>.error('Failed and it will try again');
+      } else {
+        // go to the file system and delete the file that was uploaded
+        final File file = File(inputData['imageUri'].toString());
+        file.delete();
+        return Future<bool>.value(true);
+      }
+    },
+  );
+}
 
 Future<bool> uploadCapturedPicture(
   BuildContext context, {
   required String barcode,
   required ImageField imageField,
   required Uri imageUri,
 }) async {
+  // ignore: unused_local_variable
   final AppLocalizations appLocalizations = AppLocalizations.of(context);
-  final SendImage image = SendImage(
-    lang: ProductQuery.getLanguage(),
-    barcode: barcode,
-    imageField: imageField,
-    imageUri: imageUri,
+  final Map<String, dynamic> inputData = <String, dynamic>{
+    'barcode': barcode,
+    'imageField': imageField.value,
+    'imageUri': File(imageUri.path).path,
+    'counter': 0,
+  };
+  await Workmanager().initialize(
+    callbackDispatcher,
+    // The top level function, aka callbackDispatcher
   );
-  final Status? result = await LoadingDialog.run<Status>(
-    context: context,
-    future: OpenFoodAPIClient.addProductImage(
-      ProductQuery.getUser(),
-      image,
+  // generate a random 4 digit word as the task name
```

Comment:
Suggestions:
* nextInt(1000000000)
* timestamp
* both

Reply:
+1 And maybe make this random algorithm available broadly.
Not only in this class

Reply:
is this necessary 
`final String uniqueId =   'ImageUploader_${barcode}_${imageField.value}${Random().nextInt(100)}';`
Isn't it already unique here ? 
@monsieurtanuki  @g123k

Reply:
@AshAman999 Not it's not. Just roll a dice twice: there's a 1/6 chance to roll the same number twice (I've just done it: 5-3-6-6)

Reply:
I meant was barcode is alrady unique right, and i am just adding another random to it, anyways created a random class in the code